torch/_utils.py

```diff
@@ -487,7 +486,12 @@ def _get_device_index(device, optional=False, allow_cpu=False) -> int:
         device_idx = device
     if device_idx is None:
         if optional:
-            device_idx = _get_current_device_index()
+            # If we are in scripting mode, call the corresponding JIT API
+            # to get the device index else call the eager API.
```

Comment:
This comment doesn't tell me much: "why do you need to call the JIT API?"

Reply:
`_get_current_device_index` calls _get_device_attr which uses lambda expressions which are not supported in JIT :'(

Reply:
See my more general comment in the main thread